notebooks/Emukit-tutorial-bayesian-optimization-context-variables.ipynb

```diff
@@ -0,0 +1,248 @@
+    "from emukit.experimental_design.model_free.random_design import RandomDesign\n",
+    "from GPy.models import GPRegression\n",
+    "from emukit.model_wrappers import GPyModelWrapper\n",
+    "from emukit.bayesian_optimization.acquisitions import ExpectedImprovement\n",
+    "from emukit.bayesian_optimization.loops import BayesianOptimizationLoop\n",
+    "from emukit.core.loop import FixedIterationsStoppingCondition"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "f, _ = branin_function()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we define the domain of the function to optimize as usual."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "parameter_space = ParameterSpace([ContinuousParameter('x1', -5, 10),\n",
+    "                                  ContinuousParameter('x2', 0, 15)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We build the model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "design = RandomDesign(parameter_space) # Collect random points\n",
+    "X = design.get_samples(10)\n",
+    "Y = f(X)\n",
+    "model_gpy = GPRegression(X,Y) # Train and wrap the model in Emukit\n",
+    "model_emukit = GPyModelWrapper(model_gpy)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And prepare the optimization object to run the loop."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "expected_improvement = ExpectedImprovement(model = model_emukit)\n",
+    "bayesopt_loop = BayesianOptimizationLoop(model = model_emukit,\n",
+    "                                         space = parameter_space,\n",
+    "                                         acquisition = expected_improvement,\n",
+    "                                         batch_size = 1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we set the number of iterations to run to 10."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true,
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "max_iter = 10"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To set a context, we just need to create a dictionary with the variables to fix and pass it to the Bayesian optimization object when running the optimization. Note that, every time we run new iterations we can set other variables to be the context."
```

Comment:
Can you start subsection here for better readability? This is where the context part starts I believe.

Reply:
Added!